### ISSUE TITLE
unable to deploy my app

### ISSUE THREAD

**Issue by dikshagaur10** [contentful/starter-gatsby-blog#7]
I am unable to deploy my app using Heroku with starter-gatsby-blog. When I am running the command git push heroku master, its not finishing and giving me following errors

TypeError: cannot read property "allContentfulBlogPost" of undefined

npm Error! code ELIFECYCLE
![screenshot 19](https://user-images.githubusercontent.com/42304602/44941156-e53e8680-ad4c-11e8-8edd-ff7f87d184ae.png)

npm Error! errno 1

below is the log from Heroku

npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/npmcache.6jsXf/_logs/2018-09-01T01_22_29_430Z-debug.log
-----> Build failed
       
       We're sorry this build is failing! You can troubleshoot common issues here:
       https://devcenter.heroku.com/articles/troubleshooting-node-deploys
       
       Some possible problems:
       
       - Node version not specified in package.json
       https://devcenter.heroku.com/articles/nodejs-support#specifying-a-node-js-version
       
       Love,
       Heroku
       
 !     Push rejected, failed to compile Node.js app.
 !     Push failed

**Comment by paul-f-maxson** [comment]
I'm getting the same error during circleCI build

**Comment by Khaledgarbaya** [comment]
Hey Folks,
This example is meant to work with specific content types

```
message: 'Cannot query field "allContentfulBlogPost" on type "RootQueryType". Did you mean "allContentfulProduct", "allContentfulCategory", "allContentfulBrand", "allContentfulAsset", or "allContentfulContentType"?',
```

You are querying `allContentfulBlogPost` which it does not exist, the only possible cases based on your data are `"allContentfulProduct", "allContentfulCategory", "allContentfulBrand", "allContentfulAsset", or "allContentfulContentType"`

**Comment by paul-f-maxson** [comment]
Makes sense, sorta confusing that the default template is set up this way.
